url9d04094941_petermikitsh_PizzaDeliverySystem_tgz-pJ8-testdrivers_ManagerReportTestDriverJ8/src/ninja/Driver.java

```diff
@@ -0,0 +1,188 @@
+package ninja;
+
+import java.util.ArrayList;
+
+import model.Address;
+import model.Order;
+
+/**
+ * A Driver will deliver Orders that have the same delivery location.
+ * 
+ * @author  Isioma Nnodum	iun4534@rit.edu
+ * 
+ */
+public class Driver extends Thread {
+	
+	private int _driverId;
+
+	/**
+	 * represents the state of this driver
+	 */
+	private boolean _running;
+
+	/**
+	 * The orders this driver is delivering
+	 */
+	private ArrayList<Order> _currentOrders;
+	
+	/**
+	 * The location to which this driver is traveling null if the driver is idle.
+	 */
+	private Address _deliveryLocation;
+	
+	/**
+	 * Indicates the time at which this driver started its journey to the
+	 *  delivery location.
+	 */
+	private long _startTime;
+
+	/**
+	 * Create a new driver
+	 */
+	public Driver(int id) {
+		
+		_driverId = id;
+		_running = false;
+		_currentOrders = new ArrayList<Order>();
+		_deliveryLocation = null;
+		
+	}
+
+	/**
+	 * Stop this driver
+	 */
+	public void deactivate() {
+		_running = false;
+
+	}
+
+	/**
+	 * Get the next list of orders going to the same location
+	 */
+	private void retreive() {
+
+		while ((0 == _currentOrders.size()) && _running) {
+			
+			_currentOrders = Kitchen.driverRetrieve();
+			
+			// Sleep for a bit.
+        	try {
+        		Thread.sleep( Kitchen.KITCHEN_THREAD_WAIT_TIME );
+        	} catch( InterruptedException ex ) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)